openhtf/util/measurements.py

```diff
@@ -126,14 +127,23 @@ def Doc(self, docstring):
     self.docstring = docstring
     return self
 
-  def WithUnits(self, units):
+  def _maybe_make_unit_desc(self, unit_desc):
+    """Return the UnitDescriptor or convert a string to one."""
+    if isinstance(unit_desc, str):
+      unit_desc = units.Unit(unit_desc)
+    if not (isinstance(unit_desc, units.UnitDescriptor) or unit_desc is None):
```

Comment:
prefer factoring the not thru, instead of `not (something or something)`.  alternatively, in the case of None, you might auto-create the NO_UNITS unit or whatever it is.